src/lib/instanceManager/createInstanceManager.js

```diff
@@ -52,8 +54,68 @@ module.exports = ({
       stagingEdgeConfigId,
       developmentEdgeConfigId,
       onBeforeEventSend,
+      useExistingAlloy,
       ...options
     }) => {
+      // If the instance is using a self-hosted alloy.js instance, we need to verify that the
+      // instance is properly loaded and configured.
+      if (useExistingAlloy) {
+        let realInstancePromise;
+
+        // Polling for the real instance asynchronously.
+        const getRealInstance = async () => {
+          // Verify that the instance is loaded.
+          try {
+            await poll(() => typeof window[name] === "function");
+            // eslint-disable-next-line no-unused-vars, unused-imports/no-unused-vars
+          } catch (err) {
+            turbine.logger.warn(
+              `Alloy instance "${name}" not found on window after 1000ms. Make sure the instance is loaded before the Launch library.`,
+            );
+            return undefined;
+          }
+
+          // Verify that the instance is also configured.
+          const instance = window[name];
+          let configured = false;
+          // getLibraryInfo only resolves after the instance is configured.
+          instance("getLibraryInfo").then(() => {
+            configured = true;
+          });
+
+          try {
+            await poll(() => configured);
+            // eslint-disable-next-line no-unused-vars, unused-imports/no-unused-vars
+          } catch (err) {
+            turbine.logger.warn(
+              `Alloy instance "${name}" was not configured after 1000ms. Make sure the instance is configured before loading the Launch library.`,
+            );
+            return undefined;
+          }
+
+          // Latching: Overwrite the proxy in the map with the real instance.
+          instanceByName[name] = instance;
```

Comment:
You set a proxy inside the instanceByName variable. But later in this file there are a few lines

```
      const instance = createCustomInstance({ name, components });
      .....
      instanceByName[name] = instance;
```

createCustomInstance returns something like this

```
     () => Promise.resolve({})
```

It seems that the return from createCustomInstance will overwrite the proxy. Is this what you want?